src/trees/MWNode.cpp

```diff
@@ -376,8 +376,13 @@ void MWNode<D>::cvTransform(int operation) {
         in_vec = out_vec;
         out_vec = tmp;
     }
+
+    double sf_prod = 1.0;
+    for (auto &s : sf) sf_prod *=s;
```

Comment:
If `sf` is used in this loop only, I'd chug the `this->getMWTree().` bla bla here and loop as `const auto & s`.